.github/workflows/ci_gitlab.yml

```diff
@@ -0,0 +1,133 @@
+name: ci_gitlab
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/ci_gitlab.yml
+      - 'modules/gitlab/**'
+      - 'examples/gitlab/**'
+  push:
+    branches: main
+    paths:
+    - "examples/gitlab/**"
+    - "modules/gitlab/**"
+  workflow_dispatch:
+
+env:
+  TF_CLI_ARGS: '-no-color'
+  TF_CLI_ARGS_destroy: '-auto-approve -refresh=false'
+  ROVER_RUNNER: true
+
+jobs:
+  ci:
+    runs-on: [ubuntu-latest]
+
+    container:
+      image: aztfmod/rover:0.13.6-2104.0803
+      options: --user 0
+
+    steps:
+      - name: checkout to repo
+        uses: actions/checkout@v2
+
+      - uses: actions/checkout@v2
+        with:
+          repository: azure/caf-terraform-landingzones
+          path: public
+
+      - name: Login azure
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+        run: |
+          CLIENT_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".clientId")
+          CLIENT_SECRET=$(echo "${AZURE_CREDENTIALS}" | jq -r ".clientSecret")
+          SUBSCRIPTION_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".subscriptionId")
+          TENANT_ID=$(echo "${AZURE_CREDENTIALS}" | jq -r ".tenantId")
+
+          az login --service-principal -u "${CLIENT_ID}" -p "${CLIENT_SECRET}" --tenant "${TENANT_ID}"
+
+          az account set -s "${SUBSCRIPTION_ID}"
+
+          echo "local user: $(whoami)"
+
+      - name: run the pipeline for provisioning launchpad
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+        run: |
+          export ARM_CLIENT_SECRET=$(echo "${AZURE_CREDENTIALS}" | jq -r ".clientSecret")
```

Comment:
A possible future enhancement is to use environment variables and only load/parse these once per workflow..

https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable